patches/tModLoader/Terraria/ModLoader/ModPlayer.cs

```diff
@@ -733,28 +734,27 @@ public class ModPlayer:ModType
 		/// <param name="b"></param>
 		/// <param name="a"></param>
 		/// <param name="fullBright"></param>
-		public virtual void DrawEffects(PlayerDrawInfo drawInfo, ref float r, ref float g, ref float b, ref float a, ref bool fullBright) {
+		public virtual void DrawEffects(PlayerDrawSet drawInfo, ref float r, ref float g, ref float b, ref float a, ref bool fullBright) {
 		}
 
 		/// <summary>
 		/// Allows you to modify the drawing parameters of the player before drawing begins.
 		/// </summary>
 		/// <param name="drawInfo"></param>
-		public virtual void ModifyDrawInfo(ref PlayerDrawInfo drawInfo) {
+		public virtual void ModifyDrawInfo(ref PlayerDrawSet drawInfo) {
 		}
 
 		/// <summary>
-		/// Allows you to modify the drawing of the player. This is done by removing from, adding to, or rearranging the list, by setting some of the layers' visible field to false, etc.
+		/// Allows you to add your own player draw layers.
+		/// Return an enumerable with the layers you want to add.
 		/// </summary>
-		/// <param name="layers"></param>
-		public virtual void ModifyDrawLayers(List<PlayerLayer> layers) {
-		}
+		public virtual IEnumerable<PlayerDrawLayer> AddDrawLayers() => Enumerable.Empty<PlayerDrawLayer>();
```

Comment:
Maybe this should disappear? All layers should be added by default, and `ModifyDrawInfo` used to control visibility?